src/IO/FileEncryptionCommon.h

```diff
@@ -101,6 +102,31 @@ class Encryptor
     size_t offset = 0;
 };
 
+
+/// File header which is stored at the beginning of encrypted files.
+struct Header
+{
+    Algorithm algorithm = Algorithm::AES_128_CTR;
+
+    /// Identifier of the key to encrypt or decrypt this file.
+    UInt64 key_id = 0;
+
+    /// Hash of the key to encrypt or decrypt this file.
+    UInt8 key_hash = 0;
+
+    InitVector init_vector;
+
+    /// The size of this header in bytes, including reserved bytes.
+    static constexpr const size_t kSize = 64;
```

Comment:
2 bytes for algorithm (I don't think we ever need more than 65536 different algorithms)
8 bytes for key ID
1 byte for key hash (seems to be enough because this hash is used only for the first check)
16 bytes for initial vector
37 bytes reserved